Document gz MIME type on Windows and macOS (#35211)
mdn/content: files/en-us/web/http/basics_of_http/mime_types/common_types/index.md

| `.eot`          | MS Embedded OpenType fonts                                                                 | `application/vnd.ms-fontobject`                                                                                                                                                                                                                                                                                                                                                                           |
| `.epub`         | Electronic publication (EPUB)                                                              | `application/epub+zip`                                                                                                                                                                                                                                                                                                                                                                                    |

> | `.gz`           | GZip Compressed Archive                                                                    | `application/gzip` is the standard, but beware that Windows and macOS upload `.gz` with MIME type `application/x-gzip`.                                                                                                                                                                                                                                                                                   |

Reviewer: So to be clear, if there is a request to GET a .gz file the response header have the header `Content-Type: application/gzip`, but you're saying that if you do a POST from a windows/macOS to upload a gzipped file the browser will include the `application/x-gzip` in the `Content-Type` of the data when it is being uploaded?

Is this browser independent? How does it work?

Either way, I don't love the words here or in the zip case. Perhaps

```suggestion
| `.gz`           | GZip Compressed Archive                                                                    | `application/gzip`. Note, Windows and macOS upload `.gz` with the non-standard MIME type `application/x-gzip`.                                                                                                                                                                                                                                                                                   |
```

Author: On the wording, I was just following the changes from merged PR #34000. More than happy to adjust that `.zip` line as well so these two notes are aligned or just change my PR and leave the `.zip` line - just let me know 👍 

As for `Content-Type` behaviour, the short answer is yes - an upload will send `application/x-gzip` from Windows or macOS for all the browsers I've tested so far (Chrome, Firefox and Safari [Mac only]). From past experience with `.zip` (#13409) and other file types, it seems the browsers are just being led by the OS.

The longer answer is that `Content-Type` for a response to `GET` of a `.gz` file depends on the server's implementation, which itself may be OS-dependent. A quick test on a Mac with a [few one-liner HTTP servers](https://gist.github.com/willurd/5720255) shows variation - Python returns `application/gzip`, Ruby returns `application/octet-stream` and Go returns `application/x-gzip`. Regardless of what a file is downloaded as, the browser/OS controls the `Content-Type` of the `POST` when uploading it. As part of that previous `.zip` ticket, I made https://jsfiddle.net/qacgoxrw/ which you can test easily - just drag a `.gz` file onto the drop target to see how your browser recognises it, e.g.:

<img width="552" alt="image" src="https://github.com/user-attachments/assets/fa98bcf3-628d-4b55-aa3b-d3c1617ad18f">

On Linux, however, at least for Ubuntu and Chrome, it will appear as the standard `application/gzip`.
